osquery/tables/system/linux/npm_packages.cpp

```diff
@@ -0,0 +1,59 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <string>
+
+#include <boost/filesystem.hpp>
+
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+
+namespace fs = boost::filesystem;
+namespace pt = boost::property_tree;
+
+namespace osquery {
+namespace tables {
+
+const std::map<std::string, std::string> PackageTopLevelKeys{
+    {"name", "name"},
+    {"version", "version"},
+    {"description", "description"},
+    {"author.name", "author"},
+    {"license", "license"}};
+
+QueryData genNPMPackages(QueryContext& context) {
+  QueryData results;
+
+  std::vector<std::string> paths;
+  resolveFilePattern("/usr/lib/node_modules/%/package.json", paths);
```

Comment:
If you have this passed in via `WHERE` clause then we can get this info from any directory on the system or fall back to global is one isn't provided.